django/db/backends/postgresql/creation.py

```diff
@@ -46,7 +46,8 @@ def _execute_create_test_db(self, cursor, parameters, keepdb=False):
                 return
             super()._execute_create_test_db(cursor, parameters, keepdb)
         except Exception as e:
-            if getattr(e.__cause__, "pgcode", "") != errorcodes.DUPLICATE_DATABASE:
+            cause = e.__cause__
+            if cause and not isinstance(cause, errors.DuplicateDatabase):
```

Comment:
I'm writing similar code and wondering if the `if cause` (which seems to prevent `None` from being used in the `isinstance` check) is needed. Shouldn't `isinstance(None, errors.DuplicateDatabase)` work fine?

In fact, doesn't this new version change the semantics and prevent exceptions without a cause from being handled by this branch?

Reply:
Good catch :dart: Thanks!